app/src/main/java/com/duckduckgo/app/browser/BrowserTabViewModel.kt

```diff
@@ -2875,6 +2875,12 @@ class BrowserTabViewModel @Inject constructor(
         }
     }
 
+    fun voiceSearchDisabled() {
+        omnibarViewState.value = currentOmnibarViewState().copy(
```

Comment:
I think this is not evaluated right away. Consider this steps:
1.) Install Build
2.) Search for something and verify you are in DDG search results page.
3.) Verify that microphone is visible
4.) Hide icon via settings
5.) Return to browser

After this the icon is still visible and only disappears after clicking on a link, opening a new tab etc.